src/pages/spawn/spawnHost/spawnHostButton/SpawnHostModal.tsx

```diff
@@ -0,0 +1,139 @@
+import { useEffect, useState } from "react";
+import { useQuery, useMutation } from "@apollo/client";
+import { useLocation } from "react-router-dom";
+import { useSpawnAnalytics } from "analytics";
+import { ConfirmationModal } from "components/ConfirmationModal";
+import {
+  formToGql,
+  getFormSchema,
+  useLoadFormSchemaData,
+  useVirtualWorkstationDefaultExpiration,
+  validateSpawnHostForm,
+  FormState,
+} from "components/Spawn/spawnHostModal";
+import { SpruceForm } from "components/SpruceForm";
+import { useToastContext } from "context/toast";
+import {
+  SpawnHostMutation,
+  SpawnHostMutationVariables,
+  GetSpawnTaskQuery,
+  GetSpawnTaskQueryVariables,
+} from "gql/generated/types";
+import { SPAWN_HOST } from "gql/mutations";
+import { GET_SPAWN_TASK } from "gql/queries";
+import { omit } from "utils/object";
+import { getString, parseQueryString } from "utils/queryString";
+
+interface SpawnHostModalProps {
+  open: boolean;
+  setOpen: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+export const SpawnHostModal: React.VFC<SpawnHostModalProps> = ({
+  open,
+  setOpen,
+}) => {
+  const dispatchToast = useToastContext();
+  const spawnAnalytics = useSpawnAnalytics();
+
+  // Handle distroId, taskId query param
+  const { search } = useLocation();
+  const queryParams = parseQueryString(search);
+  const taskIdQueryParam = getString(queryParams.taskId);
+  const distroIdQueryParam = getString(queryParams.distroId);
```

Comment:
Seeing this makes me excited to bring in the `useQueryParam` hooks we have In parsley.